go/api/handlers/photo.go

```diff
@@ -0,0 +1,185 @@
+// @Produce		json
+// @Security		BearerAuth
+// @Param			id		path		string	true	"ID of the trip"
+// @Param			title		formData	string	true	"Title of the photo"
+// @Param			description	formData	string	false	"Description of the photo"
+// @Param			photo	formData	file	true	"Photo file"
+// @Success		200		{object}	responses.PhotoResponse
+// @Failure		400		{object}	error
+// @Failure		401		{object}	error
+// @Router			/trips/{id}/photos [post]
+func (handler *PhotoHandler) CreatePhoto(context *gin.Context) {
+	context.Request.Body = http.MaxBytesReader(context.Writer, context.Request.Body, 10<<20)
+
+	tripIdStr := context.Param("id")
+	tripId, _ := strconv.ParseUint(tripIdStr, 10, 32)
+
+	title := context.PostForm("title")
+	description := context.PostForm("description")
+
+	file, errFile := context.FormFile("photo")
+	if errFile != nil {
+		logger.ApiWarning(context, "Photo is missing on request")
+		context.JSON(http.StatusBadRequest, gin.H{"error": "Photo is missing on request"})
+	}
+
+	if file.Size > 5<<20 {
+		logger.ApiWarning(context, "File size exceeds limit of 5 MB")
+		context.JSON(http.StatusBadRequest, gin.H{"error": "File size exceeds limit of 5 MB"})
+		return
+	}
+
+	filePath, errFilePath := utils.SaveUploadedFile(file, "photo", title)
+	if errFilePath != nil {
+		logger.ApiError(context, "Error saving file: "+filePath)
+		context.JSON(http.StatusInternalServerError, gin.H{"error": "Error saving file"})
+		return
+	}
+
+	var photo = models.Photo{
+		Title:       title,
+		Description: description,
+		TripID:      uint(tripId),
+		Path:        filePath,
+	}
+
+	err := handler.Repository.Create(&photo)
+	if err != nil {
+		errorHandlers.HandleGormErrors(err, context)
+		return
+	}
+
+	photoResponse := responses.PhotoResponse{
+		ID:          photo.ID,
+		Title:       photo.Title,
+		Description: photo.Description,
+		Path:        photo.Path,
+		CreatedAt:   photo.CreatedAt.Format(time.RFC3339),
+		UpdateAt:    photo.UpdatedAt.Format(time.RFC3339),
+	}
+
+	context.JSON(200, photoResponse)
+	logger.ApiInfo(context, "Create a photo "+photo.Path+" on trip "+tripIdStr)
+}
+
+// @Summary      Download a photo
+// @Description  Download a photo
+// @Tags         photo
+// @Accept       json
+// @Produce      application/octet-stream
+// @Security     BearerAuth
+// @Param        id          path      string  true  "ID of the trip"
+// @Param        photoID  path      string  true  "ID of the photo"
+// @Success      200         {file}    true    "A file stream of the photo"
+// @Failure      400         {object}  error
+// @Failure      401         {object}  error
+// @Failure      404         {object}  error
+// @Router       /trips/{id}/photos/{photoID}/download [get]
+func (handler *PhotoHandler) DownloadPhoto(context *gin.Context) {
+	photoID := context.Param("photoID")
+
+	photo, errPhoto := handler.Repository.Get(photoID)
+	if errPhoto != nil {
+		errorHandlers.HandleGormErrors(errPhoto, context)
+		return
+	}
+
```

Comment:
```suggestion
  // ICI
```

Il faut vérifier que la photo appartient bien au trip